### PR TITLE
[dev] Add a label to indicate if picked PR failed

### PR DESCRIPTION
**Pull request by naman03malhotra**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:
This change is to make the current process more solid as we are exploring a different branch strategy.

Though rare, for easy detection of the failed picked PRs, we can apply label "cherry picked failed" for us to go through and investigate what happened, rather than manually checking every CFE. We also send a notification and github comment, but that maybe overlooked. Using this label, we can add checks if we forget taking care of a picked failed PR.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Start with [this guide](https://woocorerelease.wordpress.com/2024/10/22/point-release-request-prr-end-to-end-flow-and-testing-instructions/) to setup, don't go into testing yet.
2. Add an additional label "cherry pick failed"
3. Now make some changes in the release branch of your choice and open a PRR (make sure a frozen branch is present)
4. Made similar changes in the trunk and frozen release branch to simulate a merge conflict.
5. Now approve the PRR, wait for the approve comment on github.
6. Merge your change.
7. This should trigger the workflow to pick changes, but it should fail due to conflicts.
8. Make sure the original PR you entered now has the failed label.
9. Repeat the same with the [CFE flow](https://woocorerelease.wordpress.com/2024/09/18/the-new-cfe-workflow-the-complete-picture/), step has some different labels.

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: .github/workflows/new-cfe-cherry-pick.yml [hunk 1]
```diff
@@ -235,6 +235,16 @@ jobs:
 
                       ${{ steps.cherry-pick-pr.outputs.cherry-pick-pr }}
 
+            - name: Apply label "cherry pick failed" using the newly created PR url
```
Comment (naman03malhotra) [inline .github/workflows/new-cfe-cherry-pick.yml:238]: I was thinking if we can add a check for this in the build zip workflow as well, as in we check if any of the PRs are open and have this label, we stop and take a look, wdyt?

**Comment by naman03malhotra** [comment]
[This is a great example](https://a8c.slack.com/archives/C01DT6U03HC/p1734492351365719) for this usecase.

**Review by jorgeatorres — APPROVED** [review]
LGTM.

**Comment by naman03malhotra** [comment]
This was merged as part of https://github.com/woocommerce/woocommerce/pull/54026 due to conflicts from trunk removing `cherry pick failed` label.

**Pull request merged**